### PR TITLE
improve UT for pod_restarter and pvc_cleaner

### PR DESCRIPTION
**Pull request by cofyc**
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/discovery/server/server_test.go [hunk 1]
```diff
@@ -68,7 +68,9 @@ func TestServer(t *testing.T) {
 		if len(pdMemberInfos.Members) <= 0 {
 			return nil, fmt.Errorf("no members yet")
 		}
-		return pdMemberInfos, nil
+		// as pdMemberInfos.Members maybe modified, we must return a copy
+		ret := *pdMemberInfos
+		return &ret, nil
```
Comment (cofyc) [inline pkg/discovery/server/server_test.go:73]: fix a data race

```
2020-08-11T06:40:00.3950334Z WARNING: DATA RACE
2020-08-11T06:40:00.3950675Z Write at 0x00c0007e42a8 by goroutine 23:
2020-08-11T06:40:00.3951246Z   github.com/pingcap/tidb-operator/pkg/discovery/server.TestServer.func2()
2020-08-11T06:40:00.3951979Z       /home/runner/work/tidb-operator/tidb-operator/go/src/github.com/pingcap/tidb-operator/pkg/discovery/server/server_test.go:101 +0x623
2020-08-11T06:40:00.3952400Z 
2020-08-11T06:40:00.3952765Z Previous read at 0x00c0007e42a8 by goroutine 44:
2020-08-11T06:40:00.3953356Z   github.com/pingcap/tidb-operator/pkg/discovery.(*tidbDiscovery).Discover()
2020-08-11T06:40:00.3954292Z       /home/runner/work/tidb-operator/tidb-operator/go/src/github.com/pingcap/tidb-operator/pkg/discovery/discovery.go:110 +0xd43
2020-08-11T06:40:00.3955023Z   github.com/pingcap/tidb-operator/pkg/discovery/server.(*server).newHandler()
2020-08-11T06:40:00.3955745Z       /home/runner/work/tidb-operator/tidb-operator/go/src/github.com/pingcap/tidb-operator/pkg/discovery/server/server.go:66 +0x3c9
2020-08-11T06:40:00.3956367Z   github.com/pingcap/tidb-operator/pkg/discovery/server.(*server).newHandler-fm()
2020-08-11T06:40:00.3957088Z       /home/runner/work/tidb-operator/tidb-operator/go/src/github.com/pingcap/tidb-operator/pkg/discovery/server/server.go:54 +0x55
2020-08-11T06:40:00.3957663Z   github.com/emicklei/go-restful.(*Container).dispatch()
2020-08-11T06:40:00.3958398Z       /home/runner/work/tidb-operator/tidb-operator/go/pkg/mod/github.com/emicklei/go-restful@v2.9.5+incompatible/container.go:288 +0x1062
2020-08-11T06:40:00.3958976Z   github.com/emicklei/go-restful.(*Container).dispatch-fm()
2020-08-11T06:40:00.3959698Z       /home/runner/work/tidb-operator/tidb-operator/go/pkg/mod/github.com/emicklei/go-restful@v2.9.5+incompatible/container.go:203 +0x5f
2020-08-11T06:40:00.3960733Z   net/http.HandlerFunc.ServeHTTP()
2020-08-11T06:40:00.3961155Z       /opt/hostedtoolcache/go/1.13.15/x64/src/net/http/server.go:2036 +0x51
2020-08-11T06:40:00.3961461Z   net/http.(*ServeMux).ServeHTTP()
2020-08-11T06:40:00.3962958Z       /opt/hostedtoolcache/go/1.13.15/x64/src/net/http/server.go:2416 +0x288
2020-08-11T06:40:00.3963272Z   net/http.serverHandler.ServeHTTP()
2020-08-11T06:40:00.3963734Z       /opt/hostedtoolcache/go/1.13.15/x64/src/net/http/server.go:2831 +0xce
2020-08-11T06:40:00.3964026Z   net/http.(*conn).serve()
2020-08-11T06:40:00.3964367Z       /opt/hostedtoolcache/go/1.13.15/x64/src/net/http/server.go:1919 +0x837
2020-08-11T06:40:00.3964572Z 
2020-08-11T06:40:00.3964841Z Goroutine 23 (running) created at:
2020-08-11T06:40:00.3965505Z   github.com/pingcap/tidb-operator/pkg/discovery/server.TestServer()
2020-08-11T06:40:00.3966170Z       /home/runner/work/tidb-operator/tidb-operator/go/src/github.com/pingcap/tidb-operator/pkg/discovery/server/server_test.go:83 +0x81e
2020-08-11T06:40:00.3966463Z   testing.tRunner()
2020-08-11T06:40:00.3966785Z       /opt/hostedtoolcache/go/1.13.15/x64/src/testing/testing.go:909 +0x199
2020-08-11T06:40:00.3966971Z 
2020-08-11T06:40:00.3967239Z Goroutine 44 (running) created at:
2020-08-11T06:40:00.3967503Z   net/http.(*Server).Serve()
2020-08-11T06:40:00.3967837Z       /opt/hostedtoolcache/go/1.13.15/x64/src/net/http/server.go:2957 +0x5b5
2020-08-11T06:40:00.3968145Z   net/http/httptest.(*Server).goServe.func1()
2020-08-11T06:40:00.3968489Z       /opt/hostedtoolcache/go/1.13.15/x64/src/net/http/httptest/server.go:298 +0xb6
2020-08-11T06:40:00.3968763Z ==================
2020-08-11T06:40:00.3969174Z --- FAIL: TestServer (0.11s)
2020-08-11T06:40:00.3973239Z ##[error]    testing.go:853: race detected during execution of test
2020-08-11T06:40:00.3977119Z FAIL
```

File: pkg/manager/member/pvc_cleaner.go [hunk 2]
```diff
@@ -115,6 +115,7 @@ func (rpc *realPVCCleaner) reclaimPV(tc *v1alpha1.TidbCluster) (map[string]strin
 		if pvc.DeletionTimestamp != nil {
 			// PVC has been deleted, skip it
 			skipReason[pvcName] = skipReasonPVCCleanerPVCHasBeenDeleted
+			skipReason[pvcName] = skipReasonPVCCleanerPVCHasBeenDeleted
```
Comment (DanielZhangQD) [inline pkg/manager/member/pvc_cleaner.go:118]: Why add this?

Comment (cofyc) [reply pkg/manager/member/pvc_cleaner.go]: my mistake, removed

**Pull request merged**